crates/miden-testing/src/kernel_tests/tx/test_account.rs

```diff
@@ -40,6 +40,74 @@ use crate::{
     Auth, MockChain, TransactionContextBuilder, assert_execution_error, executor::CodeExecutor,
 };
 
+// ACCOUNT COMMITMENT TESTS
+// ================================================================================================
+
+#[test]
+pub fn compute_current_commitment() -> miette::Result<()> {
+    let account = Account::mock(
+        ACCOUNT_ID_REGULAR_PRIVATE_ACCOUNT_UPDATABLE_CODE,
+        ONE,
+        Auth::IncrNonce,
+        TransactionKernel::assembler(),
+    );
+
+    // Precompute a commitment to a changed account so we can assert it during tx script execution.
+    let mut account_clone = account.clone();
+    let key = Digest::from([1, 2, 3, 4u32]);
+    let value = Digest::from([2, 3, 4, 5u32]);
+    account_clone.storage_mut().set_map_item(2, *key, *value).unwrap();
+    let expected_commitment = account_clone.commitment();
+
+    let tx_script = format!(
+        r#"
+        use.miden::prologue
+        use.miden::account
+        use.test::account->test_account
+
+        begin
+            exec.account::get_initial_commitment
+            # => [INITIAL_COMMITMENT]
+
+            exec.account::compute_current_commitment
+            # => [CURRENT_COMMITMENT, INITIAL_COMMITMENT]
+
+            assert_eqw.err="initial and current commitment should be equal when no changes have been made"
+
+            # update a value in the storage map
+            push.{value}
+            push.{key}
+            push.2
+            # => [slot_idx = 2, KEY, VALUE]
+            call.test_account::set_map_item
+            dropw dropw dropw dropw
+            # => []
+
+            # compute the commitment which will recompute the storage commitment
+            exec.account::compute_current_commitment
+            # => [CURRENT_COMMITMENT]
+
+            push.{expected_commitment}
+            assert_eqw.err="current commitment should match expected one"
+
+            swapdw dropw dropw
```

Comment:
Maybe this test or another could also include `get_storage_commitment` to show how it can get outdated?